conformance/utils/kubernetes/helpers.go

```diff
@@ -321,16 +321,7 @@ func MeshNamespacesMustBeReady(t *testing.T, c client.Client, timeoutConfig conf
 	require.NoErrorf(t, waitErr, "error waiting for %s namespaces to be ready", strings.Join(namespaces, ", "))
 }
 
-// GatewayAndHTTPRoutesMustBeAccepted waits until:
-//  1. The specified Gateway has an IP address assigned to it.
-//  2. The route has a ParentRef referring to the Gateway.
-//  3. All the gateway's listeners have the following conditions set to true:
-//     - ListenerConditionResolvedRefs
-//     - ListenerConditionAccepted
-//     - ListenerConditionProgrammed
-//
-// The test will fail if these conditions are not met before the timeouts.
-func GatewayAndHTTPRoutesMustBeAccepted(t *testing.T, c client.Client, timeoutConfig config.TimeoutConfig, controllerName string, gw GatewayRef, routeNNs ...types.NamespacedName) string {
+func WaitForGatewayAndRoutesToBeAccepted(t *testing.T, c client.Client, timeoutConfig config.TimeoutConfig, controllerName string, gw GatewayRef, routeNNs []types.NamespacedName, routeHandler func(t *testing.T, client client.Client, timeoutConfig config.TimeoutConfig, routeName types.NamespacedName, parents []gatewayv1.RouteParentStatus, namespaceRequired bool)) string {
```

Comment:
this is a pretty big function being accepted as a parameter. would be better to have a type defined for it (something like `routeHandlerFunc()` maybe?) and then use that here and in the below functions.